src/ol/geom/flat/CircularArc.js

```diff
@@ -0,0 +1,266 @@
+  constructor(begin, middle, end) {
+    this.begin = begin;
+    this.middle = middle;
+    this.end = end;
+  }
+
+  /**
+   * Computes and returns everything necessary in order to draw this arc.
+   * @return {{middle: Point, startAngle: number, centerOfCircle: Point, clockwise: boolean, endAngle: number, end: Point, radius: number, begin: Point}}
+   */
+  drawable = () => {
+    const center = this.centerOfCircle();
+    const angles = this.angles();
+
+    return {
+      begin: this.begin,
+      middle: this.middle,
+      end: this.end,
+      centerOfCircle: center,
+      radius: this.begin.subtract(center).magnitude(),
+      startAngle: angles.startAngle,
+      endAngle: angles.endAngle,
+      clockwise: this.clockwise(angles),
+    };
+  };
+
+  /**
+   * Returns if the arc concerns a full circle.
+   * @return {boolean} True if so, false otherwise.
+   */
+  fullCircle = () => this.begin.equals(this.end);
+
+  /**
+   * Computes and returns if the arc moves in clockwise direction.
+   * @param {{startAngle: number, endAngle: number, middleAngle: number}} angles The begin, middle and end angles.
+   * @return {boolean} True if clockwise, false otherwise.
+   */
+  clockwise = (angles) => {
+    // compute the clockwise distance from start to middle and from start to end
+    const startToMiddle =
+      (angles.middleAngle - angles.startAngle + 2 * Math.PI) % (2 * Math.PI);
+    const startToEnd =
+      (angles.endAngle - angles.startAngle + 2 * Math.PI) % (2 * Math.PI);
+
+    // clockwise if in clockwise direction we reach middle before we reach end
+    return startToMiddle < startToEnd;
+  };
+
+  /**
+   * Computes and returns the angles at all three positions with the center
+   * of the circle as their origin.
+   * @return {{startAngle: number, endAngle: number, middleAngle: number}} The computed angles.
+   */
+  angles = () => {
+    if (this.fullCircle()) {
+      return {
+        startAngle: 0,
+        middleAngle: Math.PI,
+        endAngle: 2 * Math.PI,
+      };
+    }
+
+    const center = this.centerOfCircle();
+
+    return {
+      startAngle: angleFromOrigin(center, this.begin),
+      middleAngle: angleFromOrigin(center, this.middle),
+      endAngle: angleFromOrigin(center, this.end),
+    };
+  };
+
+  /**
+   * Computes and returns the center of the circle.
+   * @return {Point} The center of the circle.
+   */
+  centerOfCircle = () => {
+    if (this.fullCircle()) {
+      // three points represent a full circle
+      return new Line(this.begin, this.middle).center();
+    }
+
+    // no full circle
+    const l1 = new Line(this.begin, this.middle);
+    const l2 = new Line(this.middle, this.end);
+
+    const perpendicularL1 = new Line(
+      l1.center(),
+      l1.center().add(l1.unit().rotated90ClockWise())
+    );
+
+    const perpendicularL2 = new Line(
+      l2.center(),
+      l2.center().add(l2.unit().rotated90ClockWise())
+    );
+
+    return perpendicularL1.intersection(perpendicularL2);
```

Comment:
In dg DIALOG BGT wordt met het middelpunt van de cirkel op een andere manier berekend, maar één stap is misschien hier nuttig.

Er worden net als hier 2 lijnen gedefinieerd:
```
lijn1 = beginpunt - middenpunt
lijn2 = eindpunt - middenpunt
```
De eerste lijn is dus vanuit het midden naar het beginpunt en de tweede lijn is vanuit het midden naar het eindpunt. Als je van deze 2 lijnen de determinant neemt dan bereken je eigenlijk de oppervlakte van het parallelogram dat door de 2 lijnen gevormd wordt:

```
| lijn1.x lijn2.x |
| lijn1.y lijn2.y | = (lijn1.x * lijn2.y) - (lijn1.y * lijn2.x)
```

Als deze determinant/oppervlakte heel klein is (bijna 0), dan heb je een collineaire boog gevonden. 

In dg DIALOG BGT is het nog wat complexer: de determinant is de noemer in een deling en als deze kleiner is dan 1 dan wordt de grootte van de noemer ook nog eens vergeleken met de grootte van de teller. Ik snap dit nog niet helemaal, maar ik denk dat we iets soortgelijks met de determinant hier kunnen doen.

Overigens tekent QGis collineaire bogen ook 'gewoon' en het is open source, maar het lijkt me erg veel werk om daar in te gaan zoeken hoe zij het opgelost hebben.

Aan de andere kant is de error die op regel 142 wordt gethrowd misschien wel uniek genoeg voor collineaire bogen